applications/crossbar/src/modules/cb_rates.erl

```diff
@@ -435,3 +452,17 @@ save_processed_rates(Context, Count) ->
                   _ = crossbar_doc:save(Context, [{'publish_doc', 'false'}]),
                   lager:debug("saved up to ~b docs (took ~b ms)", [Count, wh_util:elapsed_ms(Now)])
           end).
+
+-spec rate_for_number(ne_binary(),ne_binary()) -> cb_context:context().
+rate_for_number(Context, Phonenumber) ->
+    case wh_amqp_worker:call([{<<"To-DID">>, wnm_util:normalize_number(Phonenumber)}| wh_api:default_headers(?APP_NAME, ?APP_VERSION)]
+                             ,fun wapi_rate:publish_req/1
+                             ,fun wapi_rate:resp_v/1
+                             ,10000) of
+        {'ok', Rate} ->  
+            Routines = [fun(J) -> wh_json:set_value(<<"Base-Cost">>, wh_util:to_binary(wh_util:to_float(wh_json:get_value(<<"Base-Cost">>, J))/10000), J) end
+                        ,fun(J) -> wh_json:set_value(<<"Rate">>, wh_util:to_binary(wh_util:to_float(wh_json:get_value(<<"Rate">>, J))/10000), J) end
+                       ],
+            crossbar_util:response(lists:foldl(fun(F, J) -> F(J) end, wh_json:filter(fun({K,_}) -> lists:member(K, ?NUMBER_RESP_FIELDS)  end, Rate), Routines), Context);
```

Comment:
this works but i would break down in small functions,
something like 
-> normalize_view(Rate, Context)

normalize_view(Rate, Context) ->
   crossbar_util:response(filter_view(Rate) , Context).

filter_view(Rate) ->
  normalize_fields(wh_json:filter(fun({K,_}) -> lists:member(K, ?NUMBER_RESP_FIELDS)  end, Rate))

normalize_fields(Rate) ->
   Routines = [...],
   lists:foldl(fun(F, J) -> F(J) end, Rate, Routines).

thoughts ?